### PR TITLE
static_cast Narrowing conversions

### PR DESCRIPTION
**Pull request by cugone**
Part of splitting #155 up into smaller commits

### CODE REVIEW

File: src/States/MapViewStateHelper.cpp [hunk 1]
```diff
@@ -187,8 +187,9 @@ bool validLanderSite(Tile* t)
 	}
 
 	// bleh, direct copy from Tile::distanceTo()
-	int _x = t->x() - ccLocationX(), _y = t->y() - ccLocationY();
-	float _dist = sqrt((_x * _x) + (_y * _y));
+	int _x = t->x() - ccLocationX();
+	int _y = t->y() - ccLocationY();
```
Comment (DanRStevens) [inline src/States/MapViewStateHelper.cpp]: Technically this is sneaking in an unrelated change. This should probably have been done in a separate commit. No issue with the change though.

File: src/States/PlanetSelectState.cpp [hunk 2]
```diff
@@ -36,8 +36,14 @@ class Explosion
 			mTimer.reset();
 		}
 
-		r.drawSubImageRotated(mSheet, x, y, (mFrame % 8) * 128, (mFrame / 8) * 128, 128, 128, 270.0f);
-		//r.drawSubImage(mSheet, x, y, (mFrame % 8) * 128, (mFrame / 8) * 128, 128, 128);
+		auto rasterX = static_cast<float>(x);
+		auto rasterY = static_cast<float>(y);
+		auto posX = static_cast<float>((mFrame % 8u) * 128.0f);
+		auto posY = static_cast<float>((mFrame / 8u) * 128.0f);
```
Comment (DanRStevens) [inline src/States/PlanetSelectState.cpp]: I wonder if the scope of these `static_casts` are perhaps not quite right. The inner expression appears to be an `unsigned int` `*` `float`. Maybe the cast should be wrapping the `unsigned int` portion. Alternatively, remove the `.0f` from the `128.0f`, and maybe the `u` as well, and just do the whole internal calculation as an integer.

File: src/UI/Core/TextField.cpp [hunk 3]
```diff
@@ -139,13 +139,13 @@ void TextField::onTextInput(const std::string& _s)
 
 	if (mMaxCharacters > 0 && text().length() == mMaxCharacters) { return; }
 
-	int prvLen = text().length();
+	std::size_t prvLen = text().length();
```
Comment (DanRStevens) [inline src/UI/Core/TextField.cpp]: No issue with this change. I actually prefer this sort of solution. Just want to point out that many other places currently using `static_cast` may be due to an improper type being used. We should at some point look at the uses of `static_cast` and see if a change in type would perhaps work better.

Comment (DanRStevens) [reply src/UI/Core/TextField.cpp]: Just noticed this change is also included in #174. It maybe fits better in the other PR.

File: src/States/PlanetSelectState.cpp (hunk 2 above)
Comment (cugone) [reply src/States/PlanetSelectState.cpp]: I have no idea why I did that. With proper use of parenthesis Integer-to-float promotion should take over: `auto posX = (mFrame % 8u) * 128.0f` and `auto posY = (mFrame / 8u) * 128.0f`

Comment (cugone) [reply src/States/PlanetSelectState.cpp]: Fixed

File: src/UI/Core/TextField.cpp (hunk 3 above)
Comment (cugone) [reply src/UI/Core/TextField.cpp]: Done

**Comment by cugone** [comment]
Moved `TextField.cpp` change to #174.

File: src/States/MainReportsUiState.cpp [hunk 4]
```diff
@@ -206,7 +206,7 @@ void MainReportsUiState::initialize()
 	Panels[PANEL_SPACEPORT].Name = "Space Ports";
 
 	Renderer& r = Utility<Renderer>::get();
-	setPanelRects(r.width());
+	setPanelRects(static_cast<int>(std::floor(r.width())));
```
Comment (DanRStevens) [inline src/States/MainReportsUiState.cpp]: Ahh, this caused the build to break on Clang since there is no `<cmath>` include for `std::floor`.

Comment (DanRStevens) [reply src/States/MainReportsUiState.cpp]: Btw, why was `std::floor` added here?

Comment (cugone) [reply src/States/MainReportsUiState.cpp]: I haven't the foggiest idea why I added it.

Comment (DanRStevens) [reply src/States/MainReportsUiState.cpp]: Fixed in #180.
